gui/widgets/qopenglmousetracker.h

```diff
@@ -0,0 +1,92 @@
+//
+// Created by acaramizaru on 7/25/23.
+//
+
+#ifndef SPATIALMODELEDITOR_QOPENGLMOUSETRACKER_H
+#define SPATIALMODELEDITOR_QOPENGLMOUSETRACKER_H
+
+#include <QOpenGLWidget>
+#include <QWidget>
+#include <QtOpenGL>
+
+#include "rendering/rendering.hpp"
+
+#include <QTimer>
+
+class QOpenGLMouseTracker : public QOpenGLWidget {
+  Q_OBJECT
+public:
+  QOpenGLMouseTracker(QWidget *parent = nullptr, float lineWidth = 1.0f,
+                      float lineSelectPrecision = 10.0f,
+                      QColor selectedObjectColor = QColor(255, 255, 0),
+                      float cameraFOV = 60.0f, float cameraNearZ = 0.001f,
+                      float cameraFarZ = 2000.0f, float frameRate = 30.0f);
+  ~QOpenGLMouseTracker() = default;
+
+  void SetCameraFrustum(GLfloat FOV, GLfloat width, GLfloat height,
+                        GLfloat nearZ, GLfloat farZ);
+  void SetCameraPosition(float x, float y, float z);
+  void SetCameraOrientation(float x, float y, float z);
+
+  rendering::Vector3 GetCameraPosition();
+  rendering::Vector3 GetCameraOrientation();
+
+  void addMesh(rendering::SMesh &mesh, QColor color);
+
+  void setFPS(float frameRate);
+  void setLineWidth(float lineWidth = 1.0f);
+  void setLineSelectPrecision(float lineSelectPrecision = 10.0f);
+
+  void setSelectedObjectColor(QColor color = QColor(255, 255, 0));
+
+  QColor getSelectedObjectColor();
+
+  // colour of pixel at last mouse click position
+  [[nodiscard]] const QRgb &getColour() const;
```

Comment:
QRgb is just a uint, so this can be returned by value instead of const ref - also because if we return a reference its value may change later on, which at least to me would be surprising behaviour

Reply:
You are right... I was trying to maintain the same API from "qlabelmousetracker.hpp"...